core/model/modx/processors/resource/sort.class.php

```diff
@@ -39,6 +40,8 @@ public function process() {
         $this->getNodesFormatted($data, $this->getProperty('parent', 0));
 
         $this->fireBeforeSort();
+        $resourceBeforeSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
```

Comment:
Just realized, that's true - bit confusing name for the processor then... I will provide a check whether $this->getProperty('source_pk') is a valid resource by checking whether $this->getProperty('source_type')==="modContext"